inc/upgrade/namespace.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace HM\Cavalcade\Plugin\Upgrade;
+
+use HM\Cavalcade\Plugin as Cavalcade;
+use const HM\Cavalcade\Plugin\DATABASE_VERSION;
+
+/**
+ * Update the Cavalcade database version if required.
+ *
+ * Checks the Cavalcade database version and runs the
+ * upgrade routines as required.
+ *
+ * @return bool False if upgrade not required, true if run.
+ */
+function upgrade_database() {
+	$database_version = (int) get_site_option( 'cavalcade_db_version' );
+
+	if ( $database_version === DATABASE_VERSION ) {
+		// No upgrade required.
+		return false;
+	}
+
+	if ( $database_version < 2 ) {
+		upgrade_database_2();
+	}
+
+	update_site_option( 'cavalcade_db_version', DATABASE_VERSION );
+
+	wp_cache_delete( 'jobs', 'cavalcade-jobs' );
+
+	// Upgrade successful.
+	return true;
+}
+
+/**
+ * Upgrade Cavalcade database tables to version 2.
+ *
+ * Add and populate the `schedule` column in the jobs table.
+ */
+function upgrade_database_2() {
+	global $wpdb;
+
+	$query = "ALTER TABLE `{$wpdb->base_prefix}cavalcade_jobs`
+			  ADD `schedule` varchar(255) DEFAULT NULL
+			  AFTER `interval`";
```

Comment:
I think, although I am uncertain, that adding new columns into the middle of the table causes horrific performance problems. IIRC, it has to create a temporary table, copy all records, then replace the real one, locking the table the whole time.

Trying to verify that now, but given we don't really care about the order, should we just add this to the end? IIRC, the performance is much better with that.

Reply:
I've moved it to the end in 01ab37900611ef6ecac1d2af9ad53b64b2da3f23.